src/main/java/org/openmrs/maven/plugins/utility/DistroHelper.java

```
@@ -0,0 +1,254 @@
+                ),
+                executionEnvironment(mavenProject, mavenSession, pluginManager)
+        );
+
+        DistroProperties distroProperties = null;
+        File file = new File(path, artifact.getDestFileName());
+        ZipFile zipFile = null;
+        try {
+            zipFile = new ZipFile(file);
+
+            Enumeration<? extends ZipEntry> entries = zipFile.entries();
+
+            while(entries.hasMoreElements()){
+                ZipEntry zipEntry = entries.nextElement();
+                if("openmrs-distro.properties".equals(zipEntry.getName())){
+                    Properties properties = new Properties();
+                    properties.load(zipFile.getInputStream(zipEntry));
+                    distroProperties = new DistroProperties(properties);
+                }
+            }
+
+            zipFile.close();
+
+
+        } catch (IOException e) {
+            throw new RuntimeException("Could not read " + file.toString(), e);
+        } finally {
+            IOUtils.closeQuietly(zipFile);
+            file.delete();
+        }
+
+        return distroProperties;
+    }
+
+    public DistroProperties downloadDistroProperties(File serverPath, Server server) throws MojoExecutionException {
+        Artifact artifact = new Artifact(server.getDistroArtifactId(), server.getVersion(), server.getDistroGroupId(), "jar");
+        return downloadDistroProperties(serverPath, artifact);
+    }
+
+    /**
+     * Distro can be passed in two ways: either as maven artifact identifier or path to distro file
+     * Returns null if string is invalid as path or identifier
+     * @param distro
+     * @return
+     */
+    public DistroProperties retrieveDistroProperties(String distro) throws MojoExecutionException {
+        DistroProperties result;
+        result = getDistroPropertiesFromDir(new File(distro));
+        if(result==null){
+            Artifact artifact = parseDistroArtifact(distro);
+            if(artifact!=null){
```

Eliminate "if", parseDistroArtifact should throw an exception in such a case and end the execution.